agent/runner/agent.go

```diff
@@ -0,0 +1,56 @@
+package runner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+type agentResponse struct {
+	Key string `json:"key"`
+}
+
+func (s *Runner) getAgentToken(ctx context.Context, endpoint, orgID, envID, token string) (string, error) {
```

Comment:
Instead of using the environment endpoint, we grab the agent token from its own API